mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -106,6 +106,11 @@ class LogisticRegression(override val uid: String)
       case LabeledPoint(label: Double, features: Vector) => (label, features)
     }
     val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    trainOnInstances(instances, handlePersistence)
+  }
+
+  protected[spark] def trainOnInstances(instances: RDD[(Double, Vector)],
```

Comment:
Why don't we create `Dataframe` in LogisticRegressionWithLBFGS, so no code changes in `ml` package?